### PR TITLE
DOCSP-28763 FAQ for slow operations warnings

### PR DESCRIPTION
**Pull request by kennethdyer**
## Description

Adds an FAQ entry covering slow operation warnings and what to do about them.

## Staging
[FAQ](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-28763-slow-op-warnings/faq/#what-if-i-see-a-slow-operation-warning-)

## Jira

[DOCSP-28763](https://jira.mongodb.org/browse/DOCSP-28763)

## Build

[Build](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=642c8222b49106a49863e403)

### CODE REVIEW

File: source/faq.txt [hunk 1]
```diff
@@ -107,3 +107,20 @@ have more than 2 non-arbiter nodes and you must sync from a non-arbiter
 node. Use the source cluster's connection string to specify a
 :ref:`read preference <mongodb-uri>` for a non-arbiter, data-bearing
 node. 
+
+What If I see a Slow Operation Warning?
```
Comment (jeff-allen-mongo) [inline source/faq.txt:111]: [suggestion]

Not totally sure, but I think "if" should be lower case because it's a short conjunction, and "See" should be capitalized because it's a verb?

These are guidelines I found in the APA style guide: https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case

File: source/faq.txt [hunk 2]
```diff
@@ -107,3 +107,20 @@ have more than 2 non-arbiter nodes and you must sync from a non-arbiter
...
+or a slow write operation on the destination cluster.  The warning may
+indicate network congestion or resource strain on the source or destination
+cluster.
+
+Though the warning itself is harmless, slow operations can cause operation
+timeout errors in ``mongosync`` and migration failures.
+
+Check CPU, memory, and network usage on the source and destination clusters, 
+if the servers are underprovisioned for your needs, consider upgrading to 
+better hardware.
```
Comment (jeff-allen-mongo) [inline source/faq.txt:125]: [suggestion]

Consider making this into two separate sentences:

If you see slow operation warnings, check CPU, memory, and network usage on the source and destination clusters. If the clusters are underprovisioned for your needs, consider upgrading the cluster hardware.

Comment (jeff-allen-mongo) [inline source/faq.txt:125]: [nit]

"upgrading" already implies "better", so I think we can shorten this to "consider upgrading the cluster hardware".

File: source/faq.txt [hunk 3]
```diff
@@ -107,3 +107,20 @@ have more than 2 non-arbiter nodes and you must sync from a non-arbiter
...
+What If I see a Slow Operation Warning?
+---------------------------------------
+
+Slow operation warnings can occur during the initial sync or the application 
+of a change event when there is a slow read operation on the source cluster 
+or a slow write operation on the destination cluster.  The warning may
+indicate network congestion or resource strain on the source or destination
+cluster.
+
+Though the warning itself is harmless, slow operations can cause operation
```
Comment (jeff-allen-mongo) [inline source/faq.txt:120]: I'm wondering if we can clarify what "harmless" means here. Maybe:

While these warnings do not indicate failures themselves, slow operations can cause operation timeout errors timeout errors in ``mongosync`` and migration failures.

File: source/faq.txt [hunk 4]
```diff
@@ -107,3 +107,20 @@ have more than 2 non-arbiter nodes and you must sync from a non-arbiter
 node. Use the source cluster's connection string to specify a
 :ref:`read preference <mongodb-uri>` for a non-arbiter, data-bearing
 node. 
+
+What If I see a Slow Operation Warning?
+---------------------------------------
+
+Slow operation warnings can occur during the initial sync or the application
```
Comment (jeff-allen-mongo) [inline source/faq.txt:114]: [question]

Is there a threshold for "slow operations" that we can specify here, and can the user configure the threshold?

**Review by jeff-allen-mongo — CHANGES_REQUESTED** [review]
@kennethdyer thanks for this update. Good start here, I've just left a few suggestions for your consideration.

Jeff

File: source/faq.txt (hunk 1 above)
Comment (kennethdyer) [reply source/faq.txt:111]: It looks wrong to me, but I'm probably reading it as Latin in my head.  Changed to "if".

File: source/faq.txt (hunk 2 above)
Comment (kennethdyer) [reply source/faq.txt:125]: Fixed

Comment (kennethdyer) [reply source/faq.txt:125]: Fixed

File: source/faq.txt (hunk 3 above)
Comment (kennethdyer) [reply source/faq.txt:120]: Fixed

File: source/faq.txt (hunk 4 above)
Comment (kennethdyer) [reply source/faq.txt:114]: @jeff-allen-mongo, I checked with Jian Guan.  The threshold is hard coded at 1 minute and subject to change, so best not to specify it here.

**Comment by kennethdyer** [comment]
@jeff-allen-mongo, Changes applied.

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM!

**Comment by kennethdyer** [comment]
Verbal LGTM's received from Jian Guan and Alex Komyagin.

**Review by tdq45gj — APPROVED** [review]
